### PR TITLE
Add UpdateNamespace tests for DELETED namespace state

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Add `UpdateNamespace` tests for `DELETED` namespace state.

**Why?**
Improve test coverage for `DELETED` state.

**How did you test it?**
Updated unit and integration tests.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/metrics/defs.go [hunk 1]
```diff
@@ -1274,6 +1278,7 @@ var ScopeDefs = map[ServiceIdx]map[int]scopeDefinition{
 		MatchingClientDescribeTaskQueueScope:                  {operation: "MatchingClientDescribeTaskQueue", tags: map[string]string{ServiceRoleTagName: MatchingRoleTagValue}},
 		MatchingClientListTaskQueuePartitionsScope:            {operation: "MatchingClientListTaskQueuePartitions", tags: map[string]string{ServiceRoleTagName: MatchingRoleTagValue}},
 		FrontendClientDeprecateNamespaceScope:                 {operation: "FrontendClientDeprecateNamespace", tags: map[string]string{ServiceRoleTagName: FrontendRoleTagValue}},
+		FrontendClientDeleteNamespaceScope:                    {operation: "FrontendClientDeleteNamespace", tags: map[string]string{ServiceRoleTagName: FrontendRoleTagValue}},
```
Comment (wxing1292) [inline common/metrics/defs.go:1281]: @feedmeapples is migrating namespace APIs to admin side, so should this API be also moved to admin side?

**Review by yiminc — APPROVED** [review]

**Review by yiminc — CHANGES_REQUESTED** [review]
Should we consider alternative: one API to update namespace state that would work for Deprecate/Delete/Handover and any new state we may introduce in future? 
We just need to add one extra field `state` (default UNSPECIFIED) to existing [proto UpdateNamespaceInfo](https://github.com/temporalio/api/blob/a80e5aa7ff23b46ba22e074893983cc452c0b79f/temporal/api/namespace/v1/message.proto#L73)

**Comment by paulnpdev** [comment]
How about introducing a "NamespaceState" that includes a Deleted state.  This would allow us to offer a reasonable Undo capability, which would be very useful for self-service namespaces (if not required).  And it would simplify the model a lot.  You only need one interface!  Albeit with slightly more complex options.  But I think there' s a lot of bang for the buck here.  And it would make it easy to introduce new states for future enhancements in a backwardsly compatible manner.  Namespaces are critical ... maybe some others might have an opinion here about future needs: @mfateev, @samarabbas, @rylandg

**Comment by yiminc** [comment]
I assume @alexshtin agreed with my proposal as he stamped the PR to update [api PR](https://github.com/temporalio/api/pull/129/files)

**Review by yiminc — APPROVED** [review]

**Pull request merged**